src/groups/mqb/mqbblp/mqbblp_rootqueueengine.cpp

```diff
@@ -1562,6 +1569,212 @@ void RootQueueEngine::onTimer(bsls::Types::Int64 currentTimer)
     d_consumptionMonitor.onTimer(currentTimer);
 }
 
+bool RootQueueEngine::logAlarmCb(const mqbu::StorageKey& appKey,
+                                 const bool              enableLog) const
+{
+    // executed by the *QUEUE DISPATCHER* thread
+
+    // PRECONDITIONS
+    BSLS_ASSERT_SAFE(d_queueState_p->queue()->dispatcher()->inDispatcherThread(
+        d_queueState_p->queue()));
+
+    // Construct AppId from appKey
```

Comment:
You can replace the App lookup code with
`Apps::const_iterator cItApp = d_apps.findByKey2(AppKeyCount(appKey, 0));`

`AppKeyCount` is an (hopefully, not for too long) artefact of our transitioning from non-CSL to CSL way of registering Apps.  In short, only registered Apps are supposed to have storage and consumption monitoring.  And registered Apps should have `0` as the "count" .

Reply:
Replaced as proposed.